src/test/perl/path.t

```diff
@@ -290,45 +292,103 @@ rmtree if -d $basetest;
 ok(! $mc->directory_exists($basetest), "directory_exists false on missing directory");
 ok(! $mc->file_exists($basetest), "file_exists false on missing directory");
 ok(! $mc->any_exists($basetest), "any_exists false on missing directory");
+ok(! $mc->is_symlink($basetest), "is_symlink false on missing directory");
 
 ok(! $mc->directory_exists($basetestfile), "directory_exists false on missing file");
 ok(! $mc->file_exists($basetestfile), "file_exists false on missing file");
 ok(! $mc->any_exists($basetestfile), "any_exists false on missing file");
+ok(! $mc->is_symlink($basetestfile), "is_symlink false on missing file");
 
 makefile($basetestfile);
 
 ok($mc->directory_exists($basetest), "directory_exists true on created directory");
 ok($mc->any_exists($basetest), "any_exists true on created directory");
 ok(! $mc->file_exists($basetest), "file_exists false on created directory");
+ok(! $mc->is_symlink($basetest), "is_symlink false on created directory");
 
 ok(! $mc->directory_exists($basetestfile), "directory_exists false on created file");
 ok($mc->any_exists($basetestfile), "any_exists true on created file");
 ok($mc->file_exists($basetestfile), "file_exists true on created file");
+ok(! $mc->is_symlink($basetestfile), "is_symlink false on created file");
+
+# noreset=1
+verify_exception("existence tests do not reset exception/fail", "origfailure", 0, 1);
+ok($mc->_reset_exception_fail(), "_reset_exception_fail after existence tests");
 
-# Test (broken) symlink and _exsists methods
 
-ok(symlink("really_really_missing", $brokenlink), "broken symlink created");
-makefile("$basetest/tgtdir/tgtfile");
-ok(symlink("tgtdir", $dirlink), "directory symlink created");
-ok(symlink("tgtdir/tgtfile", $filelink), "file symlink created");
+=head2 symlink/hardlink creation/update/test
+
+=cut
+
+# Test symlink creation
+
+my $target_directory = "tgtdir";
+my $target_file1 = "tgtfile1";
+my $target_file2 = "$basetest/$target_directory/tgtfile2";
+makefile("$basetest/$target_file1");
+makefile($target_file2);
+my %opts;
+is($mc->symlink($target_directory, $dirlink), CHANGED, "directory symlink created");
+is($mc->symlink($target_file2, $filelink), CHANGED, "file symlink created");
+is($mc->symlink($target_file2, $filelink), SUCCESS, "file symlink already exists: nothing done");
```

Comment:
As said in https://github.com/quattor/CAF/pull/225#issuecomment-284447549, this is the plan! Just wanted to have the main things in place before tackling the details!